storage/aptosdb/src/pruner/state_store/mod.rs

```diff
@@ -163,31 +169,49 @@ impl StateStorePruner {
 
 pub fn prune_state_store(
     db: Arc<DB>,
-    least_readable_version: Version,
```

Comment:
it's just a reuse of name. after this call, the `least_readable_version` will be updated. Otherwise, why does Rust give an `mut` option to function params? What should be the use case for that since you can always create a new mutable copy with a name prefixed with `new_`.